jobs/jobs_api_go_sdk.go

```diff
@@ -0,0 +1,307 @@
+package jobs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"sort"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/repos"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func (js *JobSettingsResource) adjustTasks() {
+	js.sortTasksByKey()
+	for _, task := range js.Tasks {
+		sort.Slice(task.DependsOn, func(i, j int) bool {
+			return task.DependsOn[i].TaskKey < task.DependsOn[j].TaskKey
+		})
+		sortWebhookNotifications(task.WebhookNotifications)
+	}
+}
+
+func (js *JobSettingsResource) sortTasksByKey() {
+	sort.Slice(js.Tasks, func(i, j int) bool {
+		return js.Tasks[i].TaskKey < js.Tasks[j].TaskKey
+	})
+}
+
+func adjustTasks(cj *jobs.CreateJob) {
+	sortTasksByKey(cj)
+	for _, task := range cj.Tasks {
+		sort.Slice(task.DependsOn, func(i, j int) bool {
+			return task.DependsOn[i].TaskKey < task.DependsOn[j].TaskKey
+		})
+		sortWebhookNotifications(task.WebhookNotifications)
+	}
+}
+
+func sortTasksByKey(cj *jobs.CreateJob) {
```

Comment:
Moving these methods to Go SDK might be better.

Reply:
That is true, can do that later